Unity-Technologies/usd-unity-sdk
# (USDU-9) Left handed orientation not handled correctly

## Purpose of this PR

**Ticket/Jira #:** USDU-9



**Edit: After investigation this is an error with this specific USD file and not the import code. Changes reverted and added a unit test**

If a mesh is lefthanded orientation in the USD file, then it will be partially converted despite being lefthanded already. Fixed to check if mesh is lefthanded earlier and convert accordingly.

Before fix (compared to FBX version exported from Maya, on right):
![image](https://user-images.githubusercontent.com/2340643/97746724-37ac9680-1ac1-11eb-8d9e-012277f9619b.png)

After fix (compared to FBX version exported from Maya, on right):
![image](https://user-images.githubusercontent.com/2340643/97746770-48f5a300-1ac1-11eb-9a39-52c074b18bfd.png)


## Testing

**Functional Testing status:**



Added an automated test to compare left vs right handed import.

Manual Test:
1. Create a left handed mesh in USD (or use the one linked to the issue)
2. In Unity, run USD > Import as GameObjects
3. In the Unity Asset, under "Advanced", change the "Change Handedness" settings, refresh and check that result is as expected.

**Performance Testing status:**



Should not affect performance.

## Overall Product Risks


**Complexity:**

Low

**Halo Effect:**

Low

## Additional information

**Note to reviewers:**



**Reminder:**
